src/oatpp/core/parser/Caret.cpp

```diff
@@ -386,6 +391,9 @@ v_int64 Caret::StateSaveGuard::getSavedErrorCode() {
   v_float64 Caret::parseFloat64(){
     char* end;
     char* start = (char*)&m_data[m_pos];
+
+    parser::json::Utils::convertSingleDecimalSeparatorFromJsonToLocale((p_char8) start);
```

Comment:
We can't add this here - since `Caret` is a general-purpose class.
It's not serving JSON only.

Reply:
Could you explain where the correct entry points for JSON serialization/parsing are?

Reply:
Hey @langchr86 ,

My bad, sorry for not providing this information earlier.

So this PR is specific to JSON, so we want to keep this for JSON only. We don't want to alter the behavior of general-purpose methods like `utils::conversion::float64ToCharSequence`.

Here is where float gets serialized in JSON - https://github.com/oatpp/oatpp/blob/master/src/oatpp/parser/json/mapping/Serializer.hpp#L124

And here where float gets deserialized from JSON - 
https://github.com/oatpp/oatpp/blob/master/src/oatpp/parser/json/mapping/Deserializer.cpp#L157

My proposition is to put the fix right where it gets serialized/deserialized and if you need some helper methods - then you can use `json::Utils` class to put those there. We can use specialized templates or whatever is needed. You may also probably need a helper method to parse float from the `Caret` - you can also put that method to `json::Utils`.

Please let me know if you need more info or any help.

Regards,
Leonid

Reply:
Thank you for your input. I refactored the whole implementation to meet your suggestions. I also added more tests for the new (and old) features of parsing/serializing.

One thing that is kind of an ugly design is that in the `Deserializer::deserializeFloatXX` methods the constness of the pointer returned by `Caret::getCurrData()` is removed to be able to replace possible different decimal separators in the internally hold string data. But I see no easy/cheap way to improve this. Because the whole parsing is designed around the non-mutable `Caret` class. And an approach where we first copy the part of the string that contains the float number is expensive and complicated. Because first we would kind of parse the length of the float number (this would kind of involve "real" float parsing), then copy it and then do the replacement of decimal separators and parsing. Instead of something like that we implement the whole parsing of JSON by ourself. Or we would need to copy the whole data of `Caret` which could be a big JSON structure.

At the end my current implementation is pragmatic and should meet your requirements. I hope this is OK for your.